### PR TITLE
functionality for skipping events that return not found

### PR DESCRIPTION
**Pull request by Akyiaa**
- changed image tag to main for githubmonitor
- functionality to skip an event which returns a 404
- more functionality to skip events that return a not found status

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

File: build-images/github-webhook-monitor/cmd/main.go [hunk 1]
```diff
@@ -210,26 +217,30 @@ func buildHookRequest(id string) (*http.Request, error) {
...
+			if err == nil {
 
-		err = json.Unmarshal(b, &request)
-		if err == nil {
+				if eventType, ok := triggerMap.Events[request.Event]; ok {
+					url := eventType.EventListener
+					payload, _ := json.Marshal(request.Request.Payload)
+					webhookRequest, err := http.NewRequest("POST", url, bytes.NewReader(payload))
+					if err != nil {
+						return nil, err
```
Comment (techcobweb) [inline build-images/github-webhook-monitor/cmd/main.go:232]: I personally really dislike return statements in the middle of functions. There should really be only one return right at the end.
Doing that makes the code easier to follow. Everything should 'drop through' to the final return statement.

**Review by techcobweb — CHANGES_REQUESTED** [review]
I'd like return statements not to be in the middle of the code please

File: build-images/github-webhook-monitor/cmd/main.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ var latestDeliveryId string
 const (
 	latestIdPath                            = "/mnt/latestId"
 	SLEEP_TIME_SECONDS_BETWEEN_GITHUB_POLLS = 120
+	EVENT_ID_NOT_FOUND_ERROR_MSG            = "HTTP Status Code 404 - Event Not Found"
```
Comment (techcobweb) [inline build-images/github-webhook-monitor/cmd/main.go:37]: This is supposed to be an error string ?
Shouldn't it explain more about what's happened ?
eg:
"Github can't find the event it previously told us about. It returned 404 when we tried to get more information about the event." or something ?

Comment (techcobweb) [reply build-images/github-webhook-monitor/cmd/main.go:37]: Also, start the constant ERROR_MSG_xxxx as the most important thing is that it's an error message.

Comment (techcobweb) [reply build-images/github-webhook-monitor/cmd/main.go:37]: Great that it's not just buried in the code as a literal though.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by techcobweb — APPROVED** [review]
Much better. Thanks

**Pull request merged**